engineio/transport/polling/decoder.go

```diff
@@ -0,0 +1,128 @@
+package polling
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/googollee/go-socket.io/engineio/frame"
+)
+
+// decoder decodes frames from the reader.
+type decoder struct {
+	reader    *bufReader
+	lastFrame *frameReader
+}
+
+func newDecoder(buf []byte, r io.Reader) *decoder {
+	return &decoder{
+		reader: newBufReader(buf, r),
+	}
+}
+
+// NextFrame returns a reader to read the next frame.
+func (d *decoder) NextFrame() (frame.Type, io.Reader, error) {
+	if d.lastFrame != nil {
+		if err := d.lastFrame.Discard(); err != nil {
+			return 0, nil, err
+		}
+	}
+
+	if err := d.reader.Fill(); err != nil {
+		return 0, nil, err
+	}
+
+	next, err := d.reader.ReadByte()
+	if err != nil && err != io.EOF {
+		return 0, nil, err
+	}
+
+	if d.lastFrame != nil {
+		// Start from the 2nd frame, need to ignore the separator.
+		next, err = d.reader.ReadByte()
+		if err != nil && err != io.EOF {
+			return 0, nil, err
+		}
+	}
+
+	d.lastFrame = &frameReader{
+		reader:   d.reader,
+		finished: false,
```

Comment:
not important comment. 

`finished: false` - it is default value. it does not necessary

Reply:
Done.